internal/generator/templates/encode.go.tmpl

```diff
@@ -35,10 +35,26 @@
     }
     if v.{{ $field.Name }} != nil {
 {{- end }}
-{{- template "encode_compound_parameters.go.tmpl" dict "pkg" $scope "field" $field }}
+{{- template "encode_compound_parameters.go.tmpl" dict "pkg" $scope "field" $field "field_name" $field_name }}
 {{- if $field.Array }}
-    {{- template "encode_array_init.go.tmpl" dict "pkg" $scope "field_name" $field_name "array" $field.Array "type" $native.Type }}
+    {{- template "array_init.go.tmpl" dict "pkg" $scope "field_name" $field_name "field" $field "native" $native }}
+    {{- if $field.Array.Length }}
+        if len({{ $field_name }}) != int({{ goExpression $scope $field.Array.Length }}) {
+            return errors.New("array size does not match!")
+        }
+    {{- end }}
+    {{- if $native.IsMarshaler }}
+    for index, _ := range {{ $field_name }} {
```

Comment:
I actually implemented it like this at first, but it didn't work:
Note that I am taking the reference of `$field[index]`. I actually want to point to the original values of the list, that's why I am using the index operator. If I would use `value`, the pointer would be the same for each iteration (because the address of `value `doesn't change, it just gets new objects copied into it).
Basically what this for loop is doing it creates a slice of pointers to the original slice elements.